clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -39,7 +39,7 @@
 import com.linkedin.venice.client.store.transport.TransportClient;
```

Comment:
> Looks good to me, just the part around 400/500 error handling // cluster from both cluster and router, we send the path "/discover_cluster?storename=$storeName" at first, // if it does not work, try "/discover_cluster/$storeName" looks weird to me. Not sure why we had this different path, but I guess we can deal with it later.

I'll check this out a bit more before merging

Reply:
@sixpluszero You were right. This would have broken in production. When routers cannot handle a metadata request, they currently return 500 error code. I will handle this and make it return 400.

I will also remove the specific 400/500 check in ControllerClient